test/functional/p2p_segwit.py

```diff
@@ -198,15 +198,15 @@ def announce_block_and_wait_for_getdata(self, block, use_header, timeout=60):
             self.send_message(msg)
         else:
             self.send_message(msg_inv(inv=[CInv(MSG_BLOCK, block.sha256)]))
-            self.wait_for_getheaders()
+            self.wait_for_getheaders(timeout=timeout)
             self.send_message(msg)
-        self.wait_for_getdata([block.sha256])
+        self.wait_for_getdata([block.sha256], timeout=timeout)
```

Comment:
Nit: default `timeout=60`, but that's 60 secs for each inner function that takes a timeout, so the *total* timeout for `announce_block_and_wait_for_getdata` is 120 in this case.

Probably fine, but potentially confusing, i.e. `announce_block_and_wait_for_getdata(..., timeout=60)` can actually run for ~120 seconds. An alternative would be something like `timeout=timeout//2` on `wait_for_getheaders` and `wait_for_getdata` but that doesn't seem much better to me. More complex could be to subtract 'already spent time' from `timeout` after each inner `wait_*` call?

Reply:
Also nit: should `timeout` on `announce_block_and_wait_for_getdata` also be kwargs only?

Reply:
> Nit: default timeout=60, but that's 60 secs for each inner function that takes a timeout, so the total timeout for announce_block_and_wait_for_getdata is 120 in this case.

the first 60 second timeout would cause the test to crash and we wouldn't wait around for the next crash to happen :)

> Also nit: should timeout on announce_block_and_wait_for_getdata also be kwargs only?

don't think it's worth it since it's only a local function used in 1 file.

Reply:
> the first 60 second timeout would cause the test to crash and we wouldn't wait around for the next crash to happen :)

I meant if the first took say 59s to succeed then the second another 59s, but yeah it's not really an issue

Reply:
> > Nit: default timeout=60, but that's 60 secs for each inner function that takes a timeout, so the total timeout for announce_block_and_wait_for_getdata is 120 in this case.
> 
> the first 60 second timeout would cause the test to crash and we wouldn't wait around for the next crash to happen :)
> 
> > Also nit: should timeout on announce_block_and_wait_for_getdata also be kwargs only?
> 
> don't think it's worth it since it's only a local function used in 1 file.

Yeah, this was my intuition too. The same applies to the total timeout, I think it'd be worth if this was used outside the context of this file, but it is actually only used in a single test